rest/config.go

```diff
@@ -1439,6 +1439,21 @@ func (sc *ServerContext) _fetchAndLoadDatabase(nonContextStruct base.NonCancella
 	return true, nil
 }
 
+// stampSGVersion looks at each of the GatewayRegistry and updates the Sync Gateway version to start them on
+func (sc *ServerContext) stampSGVersion(ctx context.Context, bucketName string, registry *GatewayRegistry) error {
+	if !registry.SGVersion.Less(&sc.BootstrapContext.sgVersion) {
+		return nil
+	}
+	registry.SGVersion = sc.BootstrapContext.sgVersion
+	err := sc.BootstrapContext.setGatewayRegistry(ctx, bucketName, registry)
+	if err != nil {
+		base.WarnfCtx(ctx, "Error setting gateway registry in bucket %q: %q", base.MD(bucketName), err)
+		return err
+	}
+	base.InfofCtx(ctx, base.KeyConfig, "Updated Sync Gateway version number in bucket %q from %s to %s", base.MD(bucketName), registry.SGVersion, sc.BootstrapContext.sgVersion)
```

Comment:
Looks like you need to make a copy of registry.SGVersion prior to line 1447 for logging purposes.